### PR TITLE
[T3A3][T15-A2]Amro Shohoud

### PR DESCRIPTION
**Pull request by AmroShohoud**

### CODE REVIEW

**Comment by AmroShohoud** [comment]
Ready for Review

File: doc/UserGuide.md [hunk 1]
```diff
@@ -35,6 +35,8 @@ Format: `help`
 ### Adding a person: `add`
 Adds a person to the address book<br>
 Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...` 
+
+ADDRESS = BLOCK, STREET, UNIT, POSTALCODE
```
Comment (xpdavid) [inline doc/UserGuide.md]: Great! It's nice that you also update the user guide.
May be you could give more information about the format of address (e.g. `Address must be of the following format:`) rather then just giving the equation `ADDRESS = BLOCK, STREET, UNIT, POSTALCODE`

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -1,59 +0,0 @@
-package seedu.addressbook.data.person;
```
Comment (xpdavid) [inline src/seedu/addressbook/data/person/Address.java]: As point out in the tutorial, you should keep the `Address` class. And the `Address` class contains `Block` `Street` `Unit` `PostalCode` (Composition, you will learn it later)

File: src/seedu/addressbook/data/person/Block.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's address block in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidBlock(String)}
+ */
+
```
Comment (xpdavid) [inline src/seedu/addressbook/data/person/Block.java]: It's nice that you write the javadoc for the class.
Usually, there is not space between the javadoc and class definition block.

File: src/seedu/addressbook/data/person/Block.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+public class Block {
+
+    public static final String EXAMPLE = "311";
+    public static final String MESSAGE_BLOCK_CONSTRAINTS = "Address value should contain only numbers";
```
Comment (xpdavid) [inline src/seedu/addressbook/data/person/Block.java]: `"Address value should contain only numbers"` should also update the sentence right?

File: src/seedu/addressbook/data/person/Block.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    public static final String BLOCK_VALIDATION_REGEX = ".+";
+    
+    public final String value;
+    private boolean isPrivate;
```
Comment (xpdavid) [inline src/seedu/addressbook/data/person/Block.java]: Maybe it does't make sense for a block to be private.
Consider: A private street address, but non-private block number.

File: test/input.txt [hunk 6]
```diff
@@ -40,19 +40,19 @@
   add Valid Name p/12345 e/valid@e.mail a/valid, address t/invalid_-[.tag
 
   # should add correctly and list non private information
-  add Adam Brown p/111111 e/adam@gmail.com a/111, alpha street
+  add Adam Brown p/111111 e/adam@gmail.com a/111, alpha street, \#01-02, 1234
```
Comment (xpdavid) [inline test/input.txt]: It's nice that you also change the test case!

**Review by xpdavid — COMMENTED** [review]
I can see a lot of efforts you have put in this activity! Great work!

**Comment by xpdavid** [comment]
@AmroShohoud Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**